### PR TITLE
Hoje audio analysis

### PR DESCRIPTION
**Pull request by QuincySharp**
This is not done yet

### CODE REVIEW

File: classification/audio_analysis.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# End of importing the utilities module
+
+from classification.extractor.low_level_data_extractor import make_low_level_data_file
+from classification.classifier.profile_data_extractor import get_classifier_data
+from similarity.split_song import split_song
+from utilities.filehandler.handle_audio import get_MonoLoaded_Song
+from utilities.filehandler.handle_path import get_absolute_path
+from essentia.standard import MonoWriter
+
+from multiprocessing import Pool
```
Comment (Areedtz) [inline classification/audio_analysis.py]: Imports should be done in a "standard order" according to [pep8](https://pep8.org/#imports):

> Imports should be grouped in the following order:
> standard library imports
> related third party imports
> local application/library specific imports

File: classification/audio_analysis.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    path = get_absolute_path("{}.wav".format(segment_id))
+
+    MonoWriter(filename=path)(song_file)
+
+    make_low_level_data_file(path, song_output_file)
+
+    timbre, mood_relaxed, mood_party, mood_aggressive, mood_happy, mood_sad = get_classifier_data(song_output_file)
+
+    #removing the segments classifier data file
+    #subprocess.run("rm {}".format(song_output_file), shell=True)
```
Comment (Areedtz) [inline classification/audio_analysis.py]: Should these comments be removed?

File: classification/audio_analysis.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+from similarity.split_song import split_song
+from utilities.filehandler.handle_audio import get_MonoLoaded_Song
+from utilities.filehandler.handle_path import get_absolute_path
+from essentia.standard import MonoWriter
+
+from multiprocessing import Pool
+
+
+def process_data_and_extract_profiles(segment_id, song_file, song_output_file):
+    path = get_absolute_path("{}.wav".format(segment_id))
```
Comment (Areedtz) [inline classification/audio_analysis.py]: Are we only working with .wav files?

File: classification/audio_analysis.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import os
+import sys
+import re
+import csv
+import subprocess
+
+# Start of importing the utilities module
+sys.path.insert(0, os.path.abspath("../utilities/"))
+import utilities.get_song_id as s_id
+# End of importing the utilities module
```
Comment (Areedtz) [inline classification/audio_analysis.py]: Why are we importing the utilites module 2 different ways?

Also, why not just import the get_song_id function?

File: classification/classifier/profile_data_extractor.py [hunk 5]
```diff
@@ -3,62 +3,58 @@
...
 
-    output_file_path = data_file_name.split(".")[0] + "-model-data.json"
+    # Temp file used instead of writing to an actual file
+    temp_file = NamedTemporaryFile(delete=True)
 
     command = 'essentia_streaming_extractor_music_svm {} {} {}'.format(
-        data_file_name, output_file_path, profile_file)
+        data_file_name, temp_file.name, profile_file)
+
+    print("THIS IS THE COMMAND: " + command)
```
Comment (Areedtz) [inline classification/classifier/profile_data_extractor.py]: Do we need this print? :)

File: classification/classifier/profile_data_extractor.py [hunk 6]
```diff
@@ -3,62 +3,58 @@
...
-
+    
     mood_aggressive = highlevel['mood_aggressive']['value']
     mood_aggressive_probability = highlevel['mood_aggressive']['probability']
-
+    
     mood_happy = highlevel['mood_happy']['value']
     mood_happy_probability = highlevel['mood_happy']['probability']
-
+
```
Comment (Areedtz) [inline classification/classifier/profile_data_extractor.py]: Should probably remove the tab here

File: classification/classifier/profile_data_extractor.py [hunk 7]
```diff
@@ -3,62 +3,58 @@
...
-
+    
     mood_party = highlevel['mood_party']['value']
     mood_party_probability = highlevel['mood_party']['probability']
-
+    
     mood_aggressive = highlevel['mood_aggressive']['value']
     mood_aggressive_probability = highlevel['mood_aggressive']['probability']
-
+
```
Comment (Areedtz) [inline classification/classifier/profile_data_extractor.py]: Should probably remove the tab here

File: classification/classifier/profile_data_extractor.py [hunk 8]
```diff
@@ -3,62 +3,58 @@
...
     timbre_probability = highlevel['timbre']['probability']
 
     mood_relaxed = highlevel['mood_relaxed']['value']
     mood_relaxed_probability = highlevel['mood_relaxed']['probability']
-
+    
     mood_party = highlevel['mood_party']['value']
     mood_party_probability = highlevel['mood_party']['probability']
-
+
```
Comment (Areedtz) [inline classification/classifier/profile_data_extractor.py]: Should probably remove the tab here

File: classification/classifier/profile_data_extractor.py [hunk 9]
```diff
@@ -3,62 +3,58 @@
...
     highlevel = data['highlevel']
 
     timbre = highlevel['timbre']['value']
     timbre_probability = highlevel['timbre']['probability']
 
     mood_relaxed = highlevel['mood_relaxed']['value']
     mood_relaxed_probability = highlevel['mood_relaxed']['probability']
-
+
```
Comment (Areedtz) [inline classification/classifier/profile_data_extractor.py]: Should probably remove the tab here

File: classification/t/test_audio_analysis.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    #for i in range(len(split_song_list)):
+    song_output_file = "{}{}_{}_output.json".format(
+            output_folder_path, 0, song_id)
+        
+    segment_id, timbre, mood_relaxed, mood_party, mood_aggressive, mood_happy, mood_sad = process_data_and_extract_profiles(
+            0, 
+            split_song_list[0], 
+            song_output_file)
+
+    segment_id1, timbre1, mood_relaxed1, mood_party1, mood_aggressive1, mood_happy1, mood_sad1 = process_data_and_extract_profiles(
```
Comment (Areedtz) [inline classification/t/test_audio_analysis.py]: Is it possible to make this line shorter?

Comment (QuincySharp) [reply classification/t/test_audio_analysis.py]: I don't know

File: classification/classifier/profile_data_extractor.py (hunk 5 above)
Comment (QuincySharp) [reply classification/classifier/profile_data_extractor.py]: No

File: classification/audio_analysis.py (hunk 4 above)
Comment (QuincySharp) [reply classification/audio_analysis.py]: Fixed

File: classification/audio_analysis.py (hunk 2 above)
Comment (QuincySharp) [reply classification/audio_analysis.py]: yes

File: classification/audio_analysis.py (hunk 3 above)
Comment (QuincySharp) [reply classification/audio_analysis.py]: No this is the temp file

File: classification/t/test_audio_analysis.py (hunk 10 above)
Comment (QuincySharp) [reply classification/t/test_audio_analysis.py]: fixed

File: src/classification/classifier/profile_data_extractor.py [hunk 11]
```diff
@@ -3,25 +3,25 @@
...
-
-from utilities.filehandler.handle_path import get_absolute_path
-
+from tempfile import NamedTemporaryFile
 
 def get_classifier_data(data_file_name):
-    profile_file = get_absolute_path("utilities/ressources/"
-                                     + "timbre_moods_profile.yaml")
+    dirname = os.path.abspath(os.path.dirname(__file__))
+    profile_file = os.path.join(dirname, "../../utilities/ressources/timbre_moods_profile.yaml")
```
Comment (LukasBHolst) [inline src/classification/classifier/profile_data_extractor.py]: Why not use get_absolute_path method?

Comment (Areedtz) [reply src/classification/classifier/profile_data_extractor.py]: Yeah, why not?

**Pull request merged**